### PR TITLE
Refactor - release 2.0

### PR DESCRIPTION
**Pull request by duboisph**
Related to: https://github.com/skyscrapers/engineering/issues/218

Refactor:

- Improved documentation
- Consolidated variable names and formats to make it consistent with other modules
- Simplified naming

Additions:

- Return the autoscaling group names without creating a dependency on the resources
- Add initial lifecycle hooks variable

Breaking changes (will be added to the release notes):

- Removal of `project` and `environment` variables in favour of just `name`. If you want to keep the old naming format, just set `var.name` as `asg-<project>-<name>-<environment>` in the `bluegreen` module, and `<project>-<environment>-<name>` in the `scaling` module.
- Split some variables into blue and green. Some parameters were generic for both autoscaling groups and we've now made them specific for each of them.
  - `user_data` -> `blue_user_data` and `green_user_data`
  - `instance_type` -> `blue_instance_type` and `green_instance_type`
  - `disk_volume_size` -> `blue_disk_volume_size` and `green_disk_volume_size`
  - `disk_volume_type` -> `blue_disk_volume_type` and `green_disk_volume_type`
- Renamed `autoscaling_group_name` variable to `autoscaling_group_names` in the `scaling` module.

Note: We will also move to using launch templates instead of configs, but that'll go to the `2.1` release

### CODE REVIEW

File: scaling/variables.tf [hunk 1]
```diff
@@ -1,21 +1,25 @@
+variable "name" {
+  description = "Name of the Auto Scaling Groups"
```
Comment (iuriaranda) [inline scaling/variables.tf]: I think the description here is not accurate. Should be something like "Name of the created resources"

File: single-stack/variables.tf [hunk 2]
```diff
@@ -1,94 +1,86 @@
...
 
 variable "disk_volume_size" {
-  description = "(Optional) The size of the volume in gigabytes"
+  description = "The size of the volume in gigabytes"
   default     = "8"
 }
 
 variable "disk_volume_type" {
-  description = "(Optional) The type of the volume. Default is standard"
+  description = "The type of the volume. Default is standard"
```
Comment (Komorebi-E) [inline single-stack/variables.tf]: For Blue/Green this is now gp2, which is the AWS standard, why not in this stack?

File: README.md [hunk 3]
```diff
@@ -42,26 +46,27 @@ Terraform module to setup blue / green deployments
...
-  blue_min_size = "2"
-  blue_desired_capacity = "2"
-  green_max_size = "0"
-  green_min_size = "0"
-  green_desired_capacity = "0"
-  security_groups = []
+  source                 = "github.com/skyscrapers/terraform-bluegreen//blue-green"
+  name                   = "app-${terraform.workspace}"
+  blue_ami               = "ami-blabla"
+  blue_instance_type     = "t2.micro"
```
[the remark this answers is not in the document]
Comment (duboisph) [reply README.md]: It's just an example

File: single-stack/variables.tf (hunk 2 above)
Comment (duboisph) [reply single-stack/variables.tf]: Good catch, should be `gp2`

**Pull request merged**